### PR TITLE
CORE-8644 Apps Widget module refactor for code consistency

### PR DESCRIPTION
**Pull request by ash3rz**
This looks big, but it's a lot of really small changes.  Most of the changes are pushing things into an appearance class or using dependency injection.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
Everything here looks straightforward to me; all moving things to injection and appearances, looks like. I say :shipit:

File: de-lib/src/main/java/org/iplantc/de/apps/widgets/client/view/AppLaunchViewImpl.java [hunk 1]
```diff
@@ -105,33 +98,10 @@ void onLaunchButtonClicked(SelectEvent event) {
...
             }
         } else {
-            if(cleaned.getAppType().equalsIgnoreCase(App.EXTERNAL_APP)
-               && UserSettings.getInstance().isEnableWaitTimeMessage()) {
-                showWaitTimeNotice(cleaned, je);
-            } else {
-                launch(cleaned, je);
-            }
-
+            launch(cleaned, je);
```
Comment (slr71) [inline de-lib/src/main/java/org/iplantc/de/apps/widgets/client/view/AppLaunchViewImpl.java]: 👍

File: de-lib/src/main/java/org/iplantc/de/apps/widgets/client/presenter/AppLaunchPresenterImpl.java [hunk 2]
```diff
@@ -174,7 +174,25 @@ public void setViewDebugId(String baseID) {
...
+        hpcWaitDlgProvider.get(new AsyncCallback<HPCWaitTimeDialog>() {
+            @Override
+            public void onFailure(Throwable throwable) {}
+
+            @Override
+            public void onSuccess(HPCWaitTimeDialog dialog) {
+                dialog.addDialogHideHandler(hideEvent -> launchAnalysis(cleaned, je));
+                dialog.show();
+            }
+        });
```
Comment (slr71) [inline de-lib/src/main/java/org/iplantc/de/apps/widgets/client/presenter/AppLaunchPresenterImpl.java]: 👍

**Comment by ash3rz** [comment]
Thanks for the reviews!  I'm a little paranoid merging refactors in before a release (last time, I broke system messages), so I'll probably wait till after the release to merge this one just in case.  Plus, I already broke a ton of QA tests with Teams and dot menu stuff so I'm extra paranoid about bugs slipping through any cracks.

**Comment by ash3rz** [comment]
Now that the release is over, I'll go ahead and merge this - thanks for the reviews!

**Pull request merged**